### PR TITLE
Preliminary cleanup before porting PageType-related menus and dialogs to gtk4

### PR DESCRIPTION
**Pull request by bhennion**
This was part of #4508, to ease the reviewing process.
This PR only contains cleanups and extension of classes in util/ needed for #4508.

### CODE REVIEW

**Comment by bhennion** [comment]
A review commit-by-commit may be easier here, as the commits are independent from one another.

File: src/core/gui/toolbarMenubar/ToolMenuHandler.cpp [hunk 1]
```diff
@@ -325,8 +325,8 @@ void ToolMenuHandler::initEraserToolItem() {
 void ToolMenuHandler::signalConnectCallback(GtkBuilder* builder, GObject* object, const gchar* signalName,
                                             const gchar* handlerName, GObject* connectObject, GConnectFlags flags,
                                             ToolMenuHandler* self) {
-    string actionName = handlerName;
-    string groupName{};
+    std::string actionName = handlerName;
+    std::string groupName{};
```
Comment (rolandlo) [inline src/core/gui/toolbarMenubar/ToolMenuHandler.cpp:329]: What is the purpose of this change? It's not like `using std::string` can be removed after that, since `string` (without `std::`) appears on other places in the file as well.

**Review by rolandlo — COMMENTED** [review]
The cleanup looks good to me from what I understand. I have two questions about `GListViewIter` though.

File: src/util/include/util/GListView.h [hunk 2]
```diff
@@ -16,10 +16,15 @@
...
-struct GListView {
+template <class ListType, typename T>
+struct ListView {
     struct GListViewIter {
+        using iterator_category = std::forward_iterator_tag;
+        using value_type = T;
+        using pointer = T*;
+        using reference = T&;
+        using difference_type = std::ptrdiff_t;
+
```
[the remark this answers is not in the document]
Comment (bhennion) [reply src/util/include/util/GListView.h:27]: It is to meet the requirement for iterators (cf e.g. https://en.cppreference.com/w/cpp/iterator/iterator_traits).
With those, we can use the functions of the std::algorithm library. Otherwise, you'll get a compilation error.

File: src/util/include/util/GListView.h [hunk 3]
```diff
@@ -32,9 +37,7 @@ struct GListView {
         }
 
         constexpr friend auto operator==(GListViewIter const& lhs, GListViewIter const& rhs) -> bool {
-            auto tier = [](GList const& list) { return std::tie(list.data, list.next, list.prev); };
-            return (lhs.list == rhs.list) ||
-                   (lhs.list != nullptr && rhs.list != nullptr && tier(*lhs.list) == tier(*rhs.list));
+            return lhs.list == rhs.list;
```
[the remark this answers is not in the document]
Comment (bhennion) [reply src/util/include/util/GListView.h:40]: Not quite: before, you'd get that two iterators were equal if they pointed to the same GList instance (this remains) or if they pointed to GList instances with the same predecessor and successor (removed).
I just don't see how the second case can ever be useful: it would you created a UB situation where your list has some branching off.
Imagine you have two GList* A,B with A->successor == B->successor and A->predecessor == B->predecessor == nullptr. In this situation, what should A->successor->predecessor be? It could be either A or B. If it's B, then std::previous(std::next(A)) == B != A. That's UB.

File: src/core/gui/toolbarMenubar/ToolMenuHandler.cpp (hunk 1 above)
Comment (bhennion) [reply src/core/gui/toolbarMenubar/ToolMenuHandler.cpp:329]: You're quite right here. I removed those changes.

File: src/util/include/util/GListView.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (rolandlo) [reply src/util/include/util/GListView.h:27]: I don't get a compilation error when removing those lines with g++ 12.3 (even after wiping the build folder completely), but thanks for the explanation.

**Review by rolandlo — APPROVED** [review]

**Comment by bhennion** [comment]
Thanks for the review @rolandlo!
Merging in 24 hours unless an objection is raised.

**Comment by rolandlo** [comment]
It might be better to first merge #5090.
